mapcache.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2019 Ricardo Leite. All rights reserved.
+ * Licenced under the MIT licence. See COPYING file in the project root for
+ * details.
+ */
+
+#ifndef __MAPCACHE_H_
+#define __MAPCACHE_H_
+
+#include "log.h"
+#include "size_classes.h"
+#include <sys/mman.h>
+
+#define MAPCACHE_SIZE 64
+
+struct MapCacheBin {
+private:
+    char* _block = nullptr;
+    uint32_t _blockNum = 0;
+
+public:
+    char* Alloc();
+    void Flush();
+};
+
+inline char* MapCacheBin::Alloc()
+{
+    if (_blockNum == 0) {
+        _block = (char*)mmap(nullptr, SB_SIZE * MAPCACHE_SIZE, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANON, -1, 0);
```

Comment:
It could use `PageAlloc`, seeing as it is really the same

Reply:
Done